Ext/ Tor outside sensible paramters crash

**Describe the bug**
Calling ext/ tor with a large index causes error - instead of returing s.th. trivial

**To Reproduce**
```
R, (x, y, z) = polynomial_ring(QQ, ["x", "y", "z"])
A = R[x; y]
B = R[x^2; x*y; y^2; z^4]
M = subquotient(A, B)
F = free_module(R, 1)
Q, _ = quo(F, [x*F[1]])
```
Then `tor(Q, M, 20)` or `ext(Q, M, 20)` will not work

What is/was the error? I've updated the example code you gave. But for me, tor and ext don't work at all, also for e.g. index 5
```
julia> ext(Q, M, 5)
ERROR: AssertionError: 0 < i <= ngens(F)
Stacktrace:
 [1] basis
   @ ~/Projekte/OSCAR/Oscar.jl/src/Modules/UngradedModules/FreeModElem.jl:174 [inlined]
 [2] gen
   @ ~/Projekte/OSCAR/Oscar.jl/src/Modules/UngradedModules/FreeModElem.jl:178 [inlined]
 [3] getindex(S::FreeMod{QQMPolyRingElem}, i::Int64)
   @ AbstractAlgebra ~/.julia/packages/AbstractAlgebra/wVuNz/src/AbstractAlgebra.jl:353
 [4] ext(M::SubquoModule{QQMPolyRingElem}, N::SubquoModule{QQMPolyRingElem}, i::Int64)
   @ Oscar ~/Projekte/OSCAR/Oscar.jl/src/Modules/UngradedModules/HomologicalAlgebra.jl:593
 [5] top-level scope
   @ REPL[38]:1
```

--- a/src/Modules/UngradedModules/HomologicalAlgebra.jl
+++ b/src/Modules/UngradedModules/HomologicalAlgebra.jl
@@ -233,7 +233,7 @@ function tor(M::ModuleFP, N::ModuleFP, i::Int)
   return simplify_light(homology(lifted_resolution,i))[1]
 end
 
-simplify_light(F::FreeMod) = F
+simplify_light(F::FreeMod) = (F, identity_map(F), identity_map(F))
 
 #TODO, mF
 #  (hom lift) => hom and tensor functor

--- a/test/Modules/UngradedModules.jl
+++ b/test/Modules/UngradedModules.jl
@@ -340,6 +340,10 @@ end
   @test is_canonically_isomorphic(present_as_cokernel(Oscar._old_simplify(E2)[1]), M_coker)
   @test is_canonically_isomorphic(E3, M_coker)
   @test iszero(E4)
+
+  # Test that tor, ext don't crash outside of "sensible" arguments
+  T3 = tor(Q, M, 20)
+  E5 = ext(Q, M, 20)
 end
 
 @testset "Gröbner bases" begin

